.github/ISSUE_TEMPLATE/bug_report.yml

```diff
@@ -0,0 +1,171 @@
+    attributes:
+      label: "Neovim version"
+      description: "Output of `nvim --version`"
+      render: markdown
+      placeholder: |
+        NVIM v0.6.1
+        Build type&#58 Release
+        LuaJIT 2.1.0-beta3
+    validations:
+      required: true
+  - type: input
+    attributes:
+      label: "Operating system and version"
+      placeholder: "Linux 5.16.11-arch1-1, macOS 11.5, Windows 10"
+    validations:
+      required: true
+  - type: input
+    attributes:
+      label: "nvim-tree version"
+      description: "`cd <your-package-directory>/nvim-tree.lua ; git log --format='%h' -n 1`"
+      placeholder: |
+        nvim-tree branch, commit or tag number
+    validations:
+      required: true
+  - type: textarea
+    attributes:
+      label: "Steps to reproduce"
+      description: "Steps to reproduce using the minimal config provided below."
+      placeholder: |
+        1. `nvim -nu /tmp/nvt-min.lua`
+        2. `:NvimTreeOpen`
+        3. ...
+    validations:
+      required: true
+  - type: textarea
+    attributes:
+      label: "Expected behavior"
+      description: "A description of the behavior you expected:"
+  - type: textarea
+    attributes:
+      label: "Actual behavior"
+      description: "Observed behavior (may optionally include images, videos or a screencast)."
+  - type: textarea
+    attributes:
+      label: "Minimal config"
+      description: "Minimal(!) configuration necessary to reproduce the issue, using the latest version. Save this as `/tmp/nvt-min.lua`. If _absolutely_ necessary, add plugins and modify the nvim-tree setup at the indicated lines."
+      render: Lua
+      value: |
+        vim.cmd [[set runtimepath=$VIMRUNTIME]]
+        vim.cmd [[set packpath=/tmp/nvt-min/site]]
+        local package_root = '/tmp/nvt-min/site/pack'
+        local install_path = package_root .. '/packer/start/packer.nvim'
+        local function load_plugins()
+        require('packer').startup {
+          {
+            'wbthomason/packer.nvim',
+            'kyazdani42/nvim-tree.lua',
+            'kyazdani42/nvim-web-devicons',
+            -- ADD PLUGINS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
+          },
+          config = {
+            package_root = package_root,
+            compile_path = install_path .. '/plugin/packer_compiled.lua',
+            display = { non_interactive = true },
+          },
+        }
+        end
+        if vim.fn.isdirectory(install_path) == 0 then
+        print("Installing nvim-tree and dependencies.")
+        vim.fn.system { 'git', 'clone', '--depth=1', 'https://github.com/wbthomason/packer.nvim', install_path }
+        end
+        load_plugins()
+        require('packer').sync()
+        vim.cmd [[autocmd User PackerComplete ++once echo "Ready!" | lua setup()]]
+        vim.opt.termguicolors = true
+        vim.opt.cursorline = true
+
+        -- MODIFY NVIM-TREE SETTINGS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
+        _G.setup = function()
```

Comment:
This could be split into a separate file, but that complicates things for the user.

Reply:
I am wondering if it should list all available options or should users put required one by themselves.

Reply:
That is a very good question.

1. All defaults specified
\- Very verbose, large file
\+ Helps highlight which defaults the user has changed

1. User's changes only
\+ Clean setup, small file
\- User must know which defaults they have changed

The README and help directs the user to use 1.

I [prefer 2](https://github.com/alex-courtis/arch/blob/master/config/nvim/lua/nvt.lua) hovewer from bug reports it appears that many user prefer 1. They are likely just continuing with the setup that they have been directed to use.

The intent of the "clean setup" is to have the user replicate their issue with few config changes. However, if they completely replace the default config with their own, we should still be able to replicate.

Thoughts @kyazdani42 ?

Reply:
i'm not too sure, i'm a bit afraid that users just skip this because it's too lengthy. We could try and see how people behave with this new issue template, if it's too much, we can simplify this part.

Reply:
Let's see but I am leaning towards reports describing reproducible steps as: "I used `setup()` function with only option `A` set to a value which results in unexpected behaviour". Other defaults are relevant I think.

Thanks for doing leg work here! It should make @kyazdani42's life way easier now.

Reply:
It will be interesting to see what users actually do with the minimal config.